src/App.js

```diff
@@ -1,61 +1,87 @@
 import React, { Component } from 'react'
 import logo from './logo.png';
 import './App.css';
-// import movieData from "./data";
-// import Movie from "./Movie";
+import Movie from "./Movie";
 import MovieProfile from "./MovieProfile"
 
-const defaultMovie = {
-  id: 1,
-  title: "Fake Movie Title",
-  poster_path: "https://image.tmdb.org/t/p/original//7G2VvG1lU8q758uOqU6z2Ds0qpA.jpg",
-  backdrop_path: "https://image.tmdb.org/t/p/original//oazPqs1z78LcIOFslbKtJLGlueo.jpg",
-  release_date: "2019-12-04",
-  overview: "Some overview that is full of buzzwords to attempt to entice you to watch this movie! Explosions! Drama! True love! Robots! A cute dog!",
-  average_rating: 6,
-  genres: [{id: 18, name: "Drama"}, {id: 42, name: "Comedy"}],
-  budget: 63000000,
-  revenue: 100853753,
-  runtime: 139,
-  tagline: "It's a movie!"
-}
-
 class App extends Component {
+  _isLoaded = false
+
   constructor() {
     super();
+
     this.state = {
-      defaultMovie
+      allMovies: [],
+      currentMovie: this.getMovie(659986), // change this once onClick is fixed, to {}
```

Comment:
Because `this.getMovie` is an asynchronous function, it's best to move that to your componentDidMount, and begin your initial state value as an empty object. This will force you to add in some conditional rendering to make sure the current movie is only rendered if there's data inside it.

Reply:
Generally it's poor practice to have async stuff in your constructor.

Reply:
I was just using that to format my MovieProfile. What I'm having trouble with is the onClick of an individual movie.